packages/ra-ui-materialui/src/button/SaveButton.tsx

```diff
@@ -57,16 +58,24 @@ export const SaveButton = <RecordType extends RaRecord = any>(
         onClick,
         mutationOptions,
         saving,
-        disabled,
+        disabled: disabledProp,
         type = 'submit',
         transform,
         variant = 'contained',
+        alwaysEnable,
         ...rest
     } = props;
     const translate = useTranslate();
     const notify = useNotify();
     const form = useFormContext();
     const saveContext = useSaveContext();
+    const { isValidating } = useFormState();
+    // Use form isDirty, isValidating and form context saving to enable or disable the save button
+    // if alwaysEnable is undefined
+    const disabled = !valueOrDefault(
```

Comment:
the double negation is hard to reason about (!!disabledProp). Can you inverse the logic to make it more readable?